var/spack/repos/builtin/packages/exasp2/package.py

```diff
@@ -0,0 +1,85 @@
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Exasp2(MakefilePackage):
+    """ExaSP2 is a reference implementation of typical linear algebra algorithms
+    and workloads for a quantum molecular dynamics (QMD) electronic structure
+    code. The algorithm is based on a recursive second-order Fermi-Operator
+    expansion method (SP2) and is tailored for density functional based
+    tight-binding calculations of material systems. The SP2 algorithm variants
+    are part of the Los Alamos Transferable Tight-binding for Energetics
+    (LATTE) code, based on a matrix expansion of the Fermi operator in a
+    recursive series of generalized matrix-matrix multiplications. It is
+    created and maintained by Co-Design Center for Particle Applications
+    (CoPA). The code is intended to serve as a vehicle for co-design by
+    allowing others to extend and/or reimplement as needed to test performance
+    of new architectures, programming models, etc."""
+
+    tags = ['proxy-app']
+
+    homepage = "https://github.com/ECP-copa/ExaSP2"
+
+    version('develop', git='https://github.com/ECP-copa/ExaSP2',
+            branch='master')
+
+    variant('mpi', default=True, description='Build With MPI Support')
+
+    depends_on('bml')
+    depends_on('blas')
+    depends_on('lapack')
+    depends_on('mpi', when='+mpi')
+    depends_on('bml@1.2.3:+mpi', when='+mpi')
+
+    build_directory = 'src'
+
+    @property
+    def build_targets(self):
+        targets = []
+        if '+mpi' in self.spec:
+            targets.append('PARALLEL=MPI')
+            targets.append('MPICC={0}'.format(self.spec['mpi'].mpicc))
+            targets.append('MPI_LIB=-L{0} -lmpi'.format(self.spec['mpi'].prefix.lib))
+            targets.append('MPI_INCLUDE=-I{0}'.format(self.spec['mpi'].prefix.include))
+        else:
+            targets.append('PARALLEL=NONE')
+        # Current ExaSP2 build system treats "OPENBLAS" as a generic default
+        targets.append('BLAS=OPENBLAS')
+        targets.append('BLAS_ROOT=' + self.spec['blas'].prefix)
```

Comment:
please don't do that. What is wrong with 
```
math_libs = spec['lapack'].libs + spec['blas'].libs
math_libs.ld_flags  // <--- put that intsead of -lopenblas
```
?
Defining [BLAS_ROOT](https://github.com/ECP-copa/ExaSP2/blob/master/src/Makefile.vanilla#L132-L135) does not set the list of libraries provided by spack, right?

Reply:
to elaborate, what you have won't work with `intel-mkl`, not sure it will work with `atlas` either.

Reply:
you can easily try, `inte-mkl` is free now. `atlas` takes some time to build, though....

Reply:
p.s. you would actually still keep `BLAS_ROOT` defined, but you need to use `link_flags` to get `-lfoo` and `search_flags` to get `-L/prefix/lib`.